concourse/client/model.py

```diff
@@ -76,11 +79,39 @@ def __init__(self, raw:dict, concourse_api, name:str):
             warning('Pipeline did not contain resource definitions: {p}'.format(p=name))
             raise ValueError()
         self.resources = map(lambda r: Resource(r, self), resources)
+        jobs = self.raw.get('jobs', None)
+        if not jobs:
+            warning(f'Pipeline {name} did not contain job definitions')
+            raise ValueError()
+        self.jobs = map(lambda j: Job(j, self), jobs)
 
     def resources_of_types(self, types):
         return filter(lambda r: r.type in types, self.resources)
 
 
+class Job(object):
+    '''
+    Wrapper around the dictionary representing a job as part of a
+    concourse.PipelineConfig
+
+    Not intended to be instantiated by users of this module
+    '''
+    @ensure_annotations
+    def __init__(self, raw: dict, pipeline: PipelineConfig):
+        self.pipeline = pipeline
+        self.concourse_api = pipeline.concourse_api
+        self.raw = raw
+        self.name = raw['name']
```

Comment:
if not using a dataclass, rather stick to using getters to be consistent with the other code in this module